Source/WebKit/UIProcess/XR/ios/PlatformXRARKit.mm

```diff
@@ -148,14 +148,32 @@
                 return;
             }
 
-            if (active.onFrameUpdate)
-                active.onFrameUpdate({ });
+            endSession();
+        });
+}
 
-            active.renderSemaphore->signal();
+void ARKitCoordinator::endSession()
+{
+    RELEASE_LOG(XR, "ARKitCoordinator::endSession");
+    ASSERT(RunLoop::isMain());
 
-            m_state = Terminating { WTFMove(active.sessionEventClient) };
-        },
-        [&](Terminating&) { });
+    RELEASE_ASSERT(std::holds_alternative<Active>(m_state));
+    Active& active = std::get<Active>(m_state);
+
+    WTF::atomicStore(&active.terminateRequested, true);
```

Comment:
and active.terminateRequested = true; 

Though, would that cause issue in the Variant definition?